app/services/doc_auth/error_generator.rb

```diff
@@ -71,7 +71,24 @@ def generate_doc_auth_errors(response_info)
       image_metric_errors = get_image_metric_errors(response_info[:image_metrics])
       return image_metric_errors.to_h unless image_metric_errors.empty?
 
+      # If we have a selfie failure (selfie does not match id), then all three fields:
+      # front, back, and selfie should fail with specific in line errors.
       liveness_enabled = response_info[:liveness_enabled]
+      selfie_error = get_selfie_error(liveness_enabled, response_info)
+      if selfie_error == Errors::SELFIE_FAILURE
+        # This returns the same sort of object that ErrorResult.to_h returns
```

Comment:
Thanks for calling this out. I think it's too much work for this PR. I think I would need to rewrite most/all of `ErrorResult` to handle having multiple `@error` and `@error_display`s.

I know the code -looks- like it should handle this, but  **all of`ErrorResult` is only built to operate on a single error object at a time.** which doesn't work here because we need different inline errors for selfie and front/back.

This is the first time we've needed to show errors on the front, back, AND selfie boxes. In the past we needed to show errors for front and back at the same time, which is handled by error objects like this one:
`MULTIPLE_FRONT_ID_FAILURES => { long_msg: MULTIPLE_FRONT_ID_FAILURES, field_msg: FALLBACK_FIELD_LEVEL, hints: true },`

The reason I can't do something like that in this PR is that the front and back boxes should show one inline error, and the selfie box needs to show a different one. If they all showed the same inline error, then something like the multiple failure object above would work.

Reply:
Oh, one final thought. We are planning on [splitting the selfie box ](https://docs.google.com/document/d/1buLZLH2xofzQP_wqhTg3s663l91c0zXldElDDPdAiyA/edit)out onto it's own page. At that point, the current implementation of `ErrorResult.to_h` will work and this code could be removed.